### PR TITLE
Split View Feature (fix #87)

### LINKED ISSUE #87: Add split view
The old site had a toggle for split view where you could read (and copy paste) the gurmukhi in a block with each other part in its own block for full shabad/ang/etc

### PR DESCRIPTION
**Pull request by bogas04**
* Add Split View Feature
* Code cleanup
* Transform renderControls to JSX
* Fix a bug where shared panktee was undefined
* Update webpack, node-sass

### CODE REVIEW

File: src/js/main.js [hunk 1]
```diff
@@ -9,6 +9,29 @@ const default_prefs = {
...
+    }
+  } else {
+    const index = prefs[key].indexOf(option);
+    prefs[key].splice(index, 1);
+  }
+  setPref(key, prefs[key]);
+}
+
+// IE 10 doesn't support toggle class
+const toggleClass = ($el, cl) => $el.classList[$el.classList.contains(cl) ? 'remove' : 'add'](cl);
```
Comment (navdeepsinghkhalsa) [inline src/js/main.js]: This is fine for now, but we are only supporting back to IE11. Any system that supports IE10 also supports IE11.

Comment (navdeepsinghkhalsa) [reply src/js/main.js]: On further review, it seems that this is used extensively. I would rather we kept `el.classList.toggle`

**Comment by bogas04** [comment]
IE11 doesn't add support for toggle either, so we need to keep this change in order to support IE11.

Once we actually migrate to a UI framework, most of this code would disappear.

screenshot_20180109-092007

**Comment by navdeepsinghkhalsa** [comment]
I saw that on MDN as well, but this page on Microsoft's Developer site indicates that IE11 does support it. It doesn't support the second parameter, but we're not using that anyway.

https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/11865865/

**Comment by bogas04** [comment]
Oh, that's interesting. Alright then, I'll use classList.toggle

File: src/js/main.js [hunk 2]
```diff
@@ -1,77 +1,114 @@
...
   el.show().select();
-  document.execCommand("copy");
+  document.execCommand('copy');
   el.blur().hide();
 });
-$("#shabad").on("click", ".share .twitter", function() {
-  let tweet = $(this).parents(".line").children("textarea").val();
-  let shortURL = '\n'+shortenURL();
+
+$('#shabad').on('click', '.share .twitter', function () {
```
Comment (bogas04) [inline src/js/main.js]: Only this part is left in jQuery, I believe.
I'm not sure what the 2nd arg to `on` does, and how it translates to `querySelector`.

```js
  document
    .querySelector('#shabad')
    .addEventListener('click', e => {
      const $node = e.currentTarget;
      if ($node.parentNode.classList.contains('share') && $node.classList.contains('twitter')) {
        // do the thing?
      }
    });
```

Comment (navdeepsinghkhalsa) [reply src/js/main.js]: `querySelector` equivalent would be `’#shabad .share .twitter’`

Comment (bogas04) [reply src/js/main.js]: I think I'll take up this change in next sprint as the details within this click listener also makes use of some jQuery features that aren't that trivial to port. Can we go ahead with this for now?

Comment (navdeepsinghkhalsa) [reply src/js/main.js]: Sounds good. I will review again during my lunch break

**Pull request merged**